src/FEValues/interface_values.jl

```diff
@@ -0,0 +1,372 @@
+and function on the interfaces of finite elements.
+
+**Arguments:**
+
+* `quad_rule_a`: an instance of a [`FaceQuadratureRule`](@ref) for element A.
+* `quad_rule_b`: an instance of a [`FaceQuadratureRule`](@ref) for element B.
+* `func_interpol_a`: an instance of an [`Interpolation`](@ref) used to interpolate the approximated function for element A.
+* `func_interpol_b`: an instance of an [`Interpolation`](@ref) used to interpolate the approximated function for element B.
+  It defaults to `func_interpol_a`.
+* `geom_interpol_a`: an optional instance of an [`Interpolation`](@ref) which is used to interpolate the geometry for element A.
+  It defaults to `func_interpol_a`.
+* `geom_interpol_b`: an optional instance of an [`Interpolation`](@ref) which is used to interpolate the geometry for element B.
+  It defaults to `func_interpol_b`.
+ 
+**associated methods:**
+
+* [`shape_value_average`](@ref)
+* [`shape_value_jump`](@ref)
+* [`shape_gradient_average`](@ref)
+* [`shape_gradient_jump`](@ref)
+
+**Common methods:**
+
+* [`reinit!`](@ref)
+* [`getnquadpoints`](@ref)
+* [`getdetJdV`](@ref)
+
+* [`shape_value`](@ref)
+* [`shape_gradient`](@ref)
+* [`shape_divergence`](@ref)
+* [`shape_curl`](@ref)
+
+* [`function_value`](@ref)
+* [`function_gradient`](@ref)
+* [`function_symmetric_gradient`](@ref)
+* [`function_divergence`](@ref)
+* [`function_curl`](@ref)
+* [`spatial_coordinate`](@ref)
+"""
+InterfaceValues
+
+struct InterfaceValues{FVA, FVB} <: AbstractValues
+    face_values_a::FVA
+    face_values_b::FVB
+    interface_transformation::InterfaceTransformation
+end
+function InterfaceValues(quad_rule_a::FaceQuadratureRule, func_interpol_a::Interpolation,
+    geom_interpol_a::Interpolation = func_interpol_a; quad_rule_b::FaceQuadratureRule = deepcopy(quad_rule_a),
+    func_interpol_b::Interpolation = func_interpol_a, geom_interpol_b::Interpolation = func_interpol_b)
+    face_values_a = FaceValues(quad_rule_a, func_interpol_a, geom_interpol_a)
+    face_values_b = FaceValues(quad_rule_b, func_interpol_b, geom_interpol_b)
+    return InterfaceValues{typeof(face_values_a), typeof(face_values_b)}(face_values_a, face_values_b, InterfaceTransformation())
+end
+
+"""
+    reinit!(iv::InterfaceValues, face_a::FaceIndex, face_b::FaceIndex, cell_a_coords::AbstractVector{Vec{dim, T}}, cell_b_coords::AbstractVector{Vec{dim, T}}, grid::AbstractGrid) where {dim, T}
+
+Update the [`FaceValues`](@ref) in the interface (A and B) using their corresponding cell coordinates and [`FaceIndex`](@ref). This involved recalculating the transformation matrix [`transform_interface_point`](@ref)
+and mutating element B's quadrature points and its [`FaceValues`](@ref) `M, N, dMdξ, dNdξ`.
+"""
+function reinit!(iv::InterfaceValues, face_a::FaceIndex, face_b::FaceIndex, cell_a_coords::AbstractVector{Vec{dim, T}}, cell_b_coords::AbstractVector{Vec{dim, T}}, grid::AbstractGrid) where {dim, T}
```

Comment:
Is it possible to implement this one by passing just the node IDs of the two cells instead of the grid? The node ids are also available in the interface cache.